src/main/java/com/google/ehub/servlets/ProfileServlet.java

```diff
@@ -0,0 +1,62 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.ehub.data.ProfileDatastore;
+import com.google.ehub.data.UserDataManager;
+import com.google.ehub.data.UserProfile;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns HTML that contains the user's profile information.**/
+@WebServlet("/profile-data")
+public class ProfileServlet extends HttpServlet {
+  ProfileDatastore profileData = new ProfileDatastore();
+  UserProfile newUser;
```

Comment:
You don't need a field `newUser`. It is only accessed in your `doGet` method so we can just move it there as a local variable. Try to limit the scope of a variable